Referencing array index using dot fails

This is caused by a bug in HCL - https://github.com/hashicorp/hcl/issues/327#issuecomment-573241540

### Sample HCL
```hcl
records = ["${aws_elasticache_cluster.mgmt.cache_nodes.0.address}"]
```

### Stacktrace
```
panic: didn't find any token of type TokenOBrack

goroutine 1 [running]:
github.com/hashicorp/hcl/v2/hclwrite.inputTokens.PartitionType(0xc0003f1660, 0x2, 0x2, 0xc000400c88, 0x2, 0x2, 0xc00000005b, 0x0, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:88 +0x580
github.com/hashicorp/hcl/v2/hclwrite.parseTraversalStep(0x13359c0, 0xc0003e0cc0, 0xc0003f1660, 0x2, 0x2, 0xc000400c88, 0x2, 0x2, 0xc0003f1660, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:408 +0xdab
github.com/hashicorp/hcl/v2/hclwrite.parseTraversal(0xc000289100, 0x5, 0x8, 0xc0003f1480, 0x9, 0x9, 0xc000400c60, 0x9, 0x9, 0xc0003f1360, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:377 +0x3da
github.com/hashicorp/hcl/v2/hclwrite.parseExpression(0x1337440, 0xc0003d21e0, 0xc0003f1360, 0xf, 0xf, 0xc000400c48, 0xf, 0xf, 0x40)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:356 +0x175
github.com/hashicorp/hcl/v2/hclwrite.parseAttribute(0xc00047f7a0, 0xc0003f1360, 0x0, 0x0, 0xc000400c48, 0x0, 0x0, 0xc0003f1240, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:260 +0xaf4
github.com/hashicorp/hcl/v2/hclwrite.parseBodyItem(0x1334580, 0xc00047f7a0, 0xc0003f1240, 0x14, 0x14, 0xc000400c30, 0x14, 0x14, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:217 +0x632
github.com/hashicorp/hcl/v2/hclwrite.parseBody(0xc00027a580, 0xc0003f06a0, 0x33, 0x33, 0xc000400b38, 0x33, 0x33, 0x0, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:193 +0x68d
github.com/hashicorp/hcl/v2/hclwrite.parseBlock(0xc0002ad7a0, 0xc0003f06a0, 0x0, 0x0, 0xc000400b38, 0x0, 0x0, 0xc0003f0340, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:329 +0xf72
github.com/hashicorp/hcl/v2/hclwrite.parseBodyItem(0x13345c0, 0xc0002ad7a0, 0xc0003f0340, 0x3e, 0x3e, 0xc000400af0, 0x3e, 0x3e, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:219 +0x4f7
github.com/hashicorp/hcl/v2/hclwrite.parseBody(0xc00027a630, 0xc0003e8000, 0x19d, 0x200, 0xc000400000, 0x19d, 0x19d, 0x0, 0x0, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:193 +0x68d
github.com/hashicorp/hcl/v2/hclwrite.parse(0xc000390900, 0x6a0, 0x8a0, 0xc00027f840, 0x3d, 0x1, 0x1, 0x0, 0xf, 0x0, ...)
        /Users/roni/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.3.0/hclwrite/parser.go:53 +0x2e4
github.com/hashicorp/hcl/v2/hclwrite.ParseConfig(...)
```

### Workaround
```hcl
aws_elasticache_cluster.mgmt.cache_nodes[0].address
```

--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,7 @@ go 1.13
 require (
 	github.com/agext/levenshtein v1.2.2 // indirect
 	github.com/bmatcuk/doublestar v1.2.2
-	github.com/hashicorp/hcl/v2 v2.3.0
+	github.com/hashicorp/hcl/v2 v2.5.1
 	github.com/mitchellh/go-wordwrap v1.0.0 // indirect
 	github.com/mitchellh/mapstructure v1.1.2
 	github.com/thoas/go-funk v0.5.0

